src/main/java/ro/msg/learning/shop/controller/TestController.java

```diff
@@ -0,0 +1,54 @@
+package ro.msg.learning.shop.controller;
+
+import lombok.AllArgsConstructor;
+import org.flywaydb.core.Flyway;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Profile;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import ro.msg.learning.shop.DummyData;
+import ro.msg.learning.shop.repository.CustomerRepository;
+import ro.msg.learning.shop.repository.LocationRepository;
+import ro.msg.learning.shop.repository.ProductCategoryRepository;
+import ro.msg.learning.shop.repository.ProductRepository;
+import ro.msg.learning.shop.repository.SupplierRepository;
+
+@RestController
+@Profile("integrationtest")
+@AllArgsConstructor
+@RequestMapping("/db")
+public class TestController {
+    private final Flyway flyway;
+    @Autowired
+    private ProductRepository productRepository;
```

Comment:
In a controller, you should use the services, not the repositories. The Repositories are part of Data Access layer in a layered architecture and the Controllers are part of Presentation layer. The Presentation layer should not directly communicate with the Data Access Layer, they should communicate using the Business Logic Layer (which means the Services). 
![image](https://github.com/ro-msg-spring-training/online-shop-katonaaron-msg/assets/127387393/2363ccae-8567-4d9a-964a-c2634fb78fa4)

Reply:
I created an IntegrationTestDBService.